### PR TITLE
gh-95065: Produce nicer deprecation messages in Argument Clinic

### PR DESCRIPTION
**Pull request by erlend-aasland**
* Issue: gh-95065

### CODE REVIEW

**Comment by erlend-aasland** [comment]
@serhiy-storchaka: this will probably cause merge conflicts with your work on deprecating keyword arguments.

**Comment by erlend-aasland** [comment]
(the Azure failure is unrelated; flakey `test_concurrent_futures` test)

File: Lib/test/clinic.test.c [hunk 1]
```diff
@@ -6436,4 +6438,193 @@ test_deprecate_positional_pos2_len3_with_kwdonly_impl(PyObject *module,
...
+    PyObject *argsbuf[1];
+    PyObject * const *fastargs;
+    Py_ssize_t nargs = PyTuple_GET_SIZE(args);
+    PyObject *a;
+
+    // Emit compiler warnings when we get to Python 3.14.
+    #if PY_VERSION_HEX >= 0x030e00C0
+    #  error \
+            "In clinic.test.c, update parameter(s) 'a' in the clinic input of" \
+            " Test() to be keyword-only."
```
Comment (serhiy-storchaka) [inline Lib/test/clinic.test.c]: I am not sure that is an improvement. `Test` is a class, not function, there is no clinic input for it (or at least not what have parameters). Even if you guess that it is a reference to a constructor, it has two constructors. I think that `Test.__new__` would be better.

Also, in clinic input to which the message refers it is named without parentheses.

Comment (erlend-aasland) [reply Lib/test/clinic.test.c]: Hm, yes for the preprocessor message it is actually best to keep the original. I am not so sure about the deprecation message though. Will it not be better for users with this improvement?

Comment (erlend-aasland) [reply Lib/test/clinic.test.c]: I reverted the preprocessor change, but kept the deprecation change, for now.

Comment (serhiy-storchaka) [reply Lib/test/clinic.test.c]: The user usually do not call `__init__` or `__new__` directly, so it is better to use the class name there. And parentheses are at the right place there. The runtime warning message is consistent with an error message raised when pass wrong keyword argument to a function/class.

Comment (erlend-aasland) [reply Lib/test/clinic.test.c]: So I'll keep that change. Marking this as resolved.

**Review by serhiy-storchaka — APPROVED** [review]

**Comment by serhiy-storchaka** [comment]
Thank you. Nice change.

**Comment by erlend-aasland** [comment]
Thanks for the review, Serhiy; highly appreciated.

**Review by AlexWaygood — APPROVED** [review]
Very nice indeed!

**Pull request merged**